### PR TITLE
Revive code coverage reports on CI

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Generate coverage reports and upload them to codecov.io and coveralls.io. The coverage generation is actually quite slow (ca. 30min) so we may want to try and speed it up (maybe https://github.com/RPGillespie6/fastcov could help, it's used by https://github.com/quinoacomputing/quinoa now which our coverage cmake config is copied from).

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: cmake/CodeCoverage.cmake [hunk 1]
```diff
@@ -9,6 +9,7 @@
 # Modifications:
 # 1) Change "Quinoa" to "SpECTRE" and "quinoa" to "spectre"
 # 2) Split lines to make commands more legible
+# 3) Silence CMake warnings on uninitialized variables in line 61
```
Comment (wthrowe) [inline cmake/CodeCoverage.cmake]: This seems like a very fragile way to identify a change.  Perhaps put an inline comment below and refer to it here?

**Review by wthrowe — CHANGES_REQUESTED** [review]
Trying to access the report gives me 500 errors.  Is something messed up there?

**Comment by nilsvu** [comment]
@wthrowe Maybe the problem is that Codecov doesn't have a report for `develop`. See here: https://app.codecov.io/gh/sxs-collaboration/spectre/pulls?page=1&state=open&order=-pullid. When you try to select the report for this PR it errors as well.

**Comment by nilsvu** [comment]
I feel like we should disable these PR comments from codecov anyway, but that's a separate decision.

**Comment by wthrowe** [comment]
Something is screwed up about their site.  If I use your link and then click on the PR report, I get a 500.  But if I then use my browser back button and click on the PR report again I get the report.  Doesn't look like a problem with this PR, though.

**Comment by nilsvu** [comment]
@kidder and @wthrowe Here's a decision to be made before merging this PR: Generating the coverage report takes ~30min. It's only a single CI job, so won't take away a lot of resources from other jobs running in parallel, but it increases the push-to-CI-completion-time. We can try speeding it up (see PR description) but I can't currently allocate the time to try that. So what do you think: Merge or no merge?

**Comment by kidder** [comment]
An alternative would be to have a separate workflow that we can manually trigger (like the mac os build) to see how well it works on a few PRs.  
But I am fine with merging it as is and seeing how it performs, and then discussing the performance in a few weeks.

File: cmake/CodeCoverage.cmake (hunk 1 above)
Comment (nilsvu) [reply cmake/CodeCoverage.cmake]: updated this to refer to `cmake_parse_arguments` directly

**Review by wthrowe — DISMISSED** [review]
Yeah, let's just turn this on globally and reevaluate if it causes trouble.

**Comment by nilsvu** [comment]
@wthrowe @kidder Added a commit with a codecov config file. It disables PR comments and configures the GitHub status checks. People can click the status check to get to the coverage report, that's better than the PR comment I think (I find it annoying).

**Review by kidder — DISMISSED** [review]

File: cmake/SetupFormaline.cmake [hunk 2]
```diff
@@ -6,9 +6,9 @@
 # directory that are tracked by Git. I.e.
 #   git ls-tree --full-tree --name-only HEAD
 set(SPECTRE_FORMALINE_LOCATIONS
-  ".clang-format;.clang-tidy;cmake;CMakeLists.txt;containers;docs;external;"
-  ".github;.gitignore;LICENSE.txt;Metadata.yaml;README.md;src;.style.yapf;"
-  "support;tests;tools;.travis;.travis.yml")
+  ".clang-format;.clang-tidy;.codecov.yaml;cmake;CMakeLists.txt;containers;"
```
Comment (wthrowe) [inline cmake/SetupFormaline.cmake]: o is after m.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: cmake/SetupFormaline.cmake (hunk 2 above)
Comment (nilsvu) [reply cmake/SetupFormaline.cmake]: fixed 👍

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**